libp2p/protocols/pubsub/rpc/messages.nim

```diff
@@ -27,52 +27,52 @@ proc expectedFields[T](t: typedesc[T], existingFieldNames: seq[string]) {.raises
     raise newException(CatchableError, $T & " fields changed, please search for and revise all relevant procs. New fields: " & $fieldNames)
 
 type
-    PeerInfoMsg* = object
-      peerId*: PeerId
-      signedPeerRecord*: seq[byte]
-
-    SubOpts* = object
-      subscribe*: bool
-      topic*: string
-
-    MessageId* = seq[byte]
-
-    Message* = object
-      fromPeer*: PeerId
-      data*: seq[byte]
-      seqno*: seq[byte]
-      topicIds*: seq[string]
-      signature*: seq[byte]
-      key*: seq[byte]
-
-    ControlMessage* = object
-      ihave*: seq[ControlIHave]
-      iwant*: seq[ControlIWant]
-      graft*: seq[ControlGraft]
-      prune*: seq[ControlPrune]
-      idontwant*: seq[ControlIWant]
-
-    ControlIHave* = object
-      topicId*: string
-      messageIds*: seq[MessageId]
-
-    ControlIWant* = object
-      messageIds*: seq[MessageId]
-
-    ControlGraft* = object
-      topicId*: string
-
-    ControlPrune* = object
-      topicId*: string
-      peers*: seq[PeerInfoMsg]
-      backoff*: uint64
-
-    RPCMsg* = object
-      subscriptions*: seq[SubOpts]
-      messages*: seq[Message]
-      control*: Option[ControlMessage]
-      ping*: seq[byte]
-      pong*: seq[byte]
+  PeerInfoMsg* = object
+    peerId*: PeerId
+    signedPeerRecord*: seq[byte]
+
+  SubOpts* = object
+    subscribe*: bool
+    topic*: string
+
+  MessageId* = seq[byte]
+
+  Message* = object
+    fromPeer*: PeerId
+    data*: seq[byte]
+    seqno*: seq[byte]
+    topicId*: string
```

Comment:
good opportunity to line up the field name with the spec protobuf name

Reply:
On the one hand, we can't change `fromPeer` to `from` as the second one is a reserved keyword in nim.

On the other, `go-lib-p2p` does use `Message.topic` ([ref](https://github.com/libp2p/go-libp2p-pubsub/blob/048a4d30d0c3c00829181fd81aa697eb60497448/pb/rpc.proto#L21)), but uses `ControlIHave.topicID` ([ref](https://github.com/libp2p/go-libp2p-pubsub/blob/048a4d30d0c3c00829181fd81aa697eb60497448/pb/rpc.proto#L34)). I don't know which one I like more, but I do know I'd stick to using one and the same for both cases (as we're already doing).

Reply:
The canonical names for each field are in the spec: 

https://github.com/libp2p/specs/blob/master/pubsub/README.md#the-message

It doesn't greatly matter what go-libp2p does, it's not authoritative. 

Technically, we can use \`from\` (in backticks), though it's a bit of a bother in nim, so for that field in particular, we have reason to not go with the canonical name).

Reply:
I did check the spec, but also double checked what other libraries were doing given the `from` situation (given I wasn't aware of the backtick trick).

I'm fing changing it to `topic` tbh, doing it.

Reply:
We will likely move to a model where we generate fields from the `.proto` directly so this is a step in the right direction. It's also useful when using the [protobuf JSON mapping](https://protobuf.dev/programming-guides/proto3/#json) - again, not for today, but good habit in general.